variables.tf

```
@@ -0,0 +1,140 @@
+variable "subnet" {
+  type = object({
+    id               = string
+    address_prefixes = list(string)
+  })
+  nullable    = false
+  description = "The variable takes the subnet as input and takes the id and the address prefix for further configuration."
+}
+
+variable "resource_group_id" {
+  type        = string
+  nullable    = false
+  description = "Id of the resource group where the resources will be created."
+}
+
+variable "tags" {
+  type        = map(string)
+  default     = {}
+  nullable    = false
+  description = "A map of tags that will be set on every resource this module creates."
+}
+
+variable "domain_configuration_type" {
+  type        = string
+  default     = "FullySynced"
+  nullable    = false
+  description = "The configuration type of this Active Directory Domain."
+  validation {
+    condition     = contains(["FullySynced", "ResourceTrusting"], var.domain_configuration_type)
+    error_message = "Possible values for the configuration type are FullySynced and ResourceTrusting."
+  }
+}
+
+variable "security_settings" {
+  type = object({
+    channelBinding        = optional(bool, true)
+    kerberosArmoring      = optional(bool, true)
+    kerberosRc4Encryption = optional(bool, false)
+    ldapSigning           = optional(bool, true)
+    ntlmV1                = optional(bool, false)
+    syncKerberosPasswords = optional(bool, true)
+    syncNtlmPasswords     = optional(bool, true)
```

```suggestion
    syncNtlmPasswords     = optional(bool, false)
```
NTML will be phased out, so unreasonable to enable on deafult.